examples/examples-cpu/nyc-taxi/dashboard.ipynb

```
@@ -1,6 +1,773 @@
+    "```\n",
+    "\n",
+    "Then your dashboard URL would be: \n",
+    "\n",
+    "```\n",
+    "https://main.demo.saturnenterprise.io/user/aaron/examples-cpu/proxy/5006/dashboard\n",
+    "```\n",
+    "\n",
+    "It will take a few seconds to load when first viewing the page, as all the cells in this notebook must be executed first.\n",
+    "\n",
+    "### Deployment \n",
+    "\n",
+    "To run as part of a Deployment, use this for the Command (see readme for more details):\n",
+    "\n",
+    "```bash\n",
+    "python -m panel serve /home/jovyan/project/examples/nyc-taxi/dashboard.ipynb --port=8000 --address=\"0.0.0.0\" --allow-websocket-origin=\"*\"\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# ML model predictions\n",
+    "\n",
+    "The `MODEL_URL` environment variable must be set to be able to get predictions from a deployed model. Otherwise the widget on the \"ML\" tab will return -1."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import datetime as dt\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "\n",
+    "import dask.dataframe as dd\n",
+    "\n",
+    "import hvplot.dask, hvplot.pandas\n",
+    "import holoviews as hv\n",
+    "from holoviews.streams import Selection1D\n",
+    "from bokeh.models import HoverTool\n",
+    "import panel as pn\n",
+    "\n",
+    "import warnings\n",
+    "warnings.simplefilter(\"ignore\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# path to aggregated data files, default uses Saturn's public S3\n",
+    "# otherwise, you can run the data_aggregation.ipynb and point to your own S3 location\n",
+    "saturn_path = 's3://saturn-public-data/nyc-taxi-dashboard'\n",
+    "DATA_PATH = os.environ.get('DASHBOARD_DATA', saturn_path)\n",
```

similar to my other comments, I don't think this "take in an environment variable or use the hard-coded Saturn path" thing is worth it. It's not really going to work unless someone uses their own bucket where they host the Taxi data exactly the way we did, since later in the code you have lines like this:

```python
{DATA_PATH}/taxi_zones/taxi_zones.shp'
```

I think this should just access an `S3_TAXI_BUCKET` (or whatever you want to call it) environment variable, and should raise a loud error if that fails. That will be a way to teach how environment variables in Saturn works, while also reducing the amount of stuff you have to think about in this code.